test/model/autocomplete_test.dart

```diff
@@ -274,35 +276,35 @@ void main() {
     }
   });
 
-  test('MentionAutocompleteView mutating store.users while in progress causes retry', () async {
-    const narrow = CombinedFeedNarrow();
+  test('MentionAutocompleteView mutating store.users while in progress does not '
+      'interrupt the current query results', () async {
+    const narrow = StreamNarrow(1);
     final store = eg.store();
-    for (int i = 0; i < 1500; i++) {
+    for (int i = 0; i < 2500; i++) {
       await store.addUser(eg.user(userId: i, email: 'user$i@example.com', fullName: 'User $i'));
     }
     final view = MentionAutocompleteView.init(store: store, narrow: narrow);
 
     bool done = false;
     view.addListener(() { done = true; });
-    view.query = MentionAutocompleteQuery('User 10000');
+    view.query = MentionAutocompleteQuery('User 110');
 
     await Future(() {});
     check(done).isFalse();
-    await store.addUser(eg.user(userId: 10000, email: 'user10000@example.com', fullName: 'User 10000'));
+    await store.addUser(eg.user(userId: 11000, email: 'user11000@example.com', fullName: 'User 11000'));
     await Future(() {});
     check(done).isFalse();
-    await Future(() {});
-    check(done).isTrue();
-    check(view.results).single
-      .isA<UserMentionAutocompleteResult>()
-      .userId.equals(10000);
-    // new result sticks; no "zombie" result from `store.users` pre-mutation
-    for (int i = 0; i < 10; i++) { // for good measure
+    for (int i = 0; i < 3; i++) {
       await Future(() {});
-      check(view.results).single
-        .isA<UserMentionAutocompleteResult>()
-        .userId.equals(10000);
     }
```

Comment:
This is OK as is because it's only 3 iterations anyway, but if it were many more I'd want to have it break as soon as the thing is done (in addition to having an upper bound on the number of iterations). See my suggested code above, or those examples I mentioned in the message-list tests, which look like:
```dart
      for (int i = 0; i < 30; i++) {
        // Find the point in the fling where the fetch starts.
        await tester.pump(const Duration(milliseconds: 100));
        if (itemCount(tester)! > 101) break; // The loading indicator appeared.
      }
```

Note the upper bound on iteration count (30) as well as the `break`.